core/src/test/scala/kafka/server/KafkaRequestHandlerTest.scala

```diff
@@ -698,4 +698,81 @@ class KafkaRequestHandlerTest {
     // cleanup
     brokerTopicStats.close()
   }
+
+  @Test
+  def testRequestThreadMetrics(): Unit = {
+    val time = Time.SYSTEM
+    val metricsBroker = new RequestChannelMetrics(java.util.Set.of[ApiKeys])
+    val metricsController = new RequestChannelMetrics(java.util.Set.of[ApiKeys])
+    val requestChannelBroker = new RequestChannel(10, time, metricsBroker)
+    val requestChannelController = new RequestChannel(10, time, metricsController)
+    val apiHandler = mock(classOf[ApiRequestHandler])
+    val metricsGroup = new KafkaMetricsGroup("kafka.server", "KafkaRequestHandlerPool")
+
+    // Create a factory for this test
+    val factory = new KafkaRequestHandlerPoolFactory()
+    
+    // Create broker pool with 4 threads
+    val brokerPool = factory.createPool(
+      0,
+      requestChannelBroker,
+      apiHandler,
+      time,
+      4,
+      "broker"
+    )
+
+    // Verify global counter is updated
+    assertEquals(4, factory.getAggregateThreads.get(), "global counter should be 4 after broker pool")
+
+    // Create controller pool with 4 threads
+    val controllerPool = factory.createPool(
+      0,
+      requestChannelController,
+      apiHandler,
+      time,
+      4,
+      "controller"
+    )
+
+    // Verify global counter is updated to sum of both pools
+    assertEquals(8, factory.getAggregateThreads.get, "global counter should be 8 after both pools")
+
+    val aggregateMeter = metricsGroup.newMeter("RequestHandlerAvgIdlePercent", "percent", TimeUnit.NANOSECONDS)
+    val brokerPerPoolIdleMeter = metricsGroup.newMeter("BrokerRequestHandlerAvgIdlePercent", "percent", TimeUnit.NANOSECONDS)
+    val controllerPerPoolIdleMeter = metricsGroup.newMeter("ControllerRequestHandlerAvgIdlePercent", "percent", TimeUnit.NANOSECONDS)
+
+    var aggregateValue = 0.0
+    var brokerPerPoolValue = 0.0
+    var controllerPerPoolValue = 0.0
+
+    // Wait until all idle-percent meters have been initialized with non-zero rates,
+    // or timeout after the given duration.
+    val startTime = time.milliseconds()
+    while(aggregateValue == 0.0 || brokerPerPoolValue == 0.0 || controllerPerPoolValue == 0.0) {
+      if (time.milliseconds() - startTime > 8000)
+        throw new RuntimeException("Timeout waiting for idle-percent metrics to initialize")
+      aggregateValue = aggregateMeter.oneMinuteRate()
+      brokerPerPoolValue = brokerPerPoolIdleMeter.oneMinuteRate()
+      controllerPerPoolValue = controllerPerPoolIdleMeter.oneMinuteRate()
+    }
```

Comment:
You need this hack because you are using `Time.SYSTEM` for the test. You can use MockTime and that object's `sleep(int)` method.